e2e/nomostest/clean.go

```diff
@@ -383,6 +390,13 @@ func DeleteObjectsAndWait(nt *NT, objs ...client.Object) error {
 		if err != nil {
 			return err
 		}
+		if testutils.RemoveFinalizer(obj, ConfigSyncE2EFinalizer) {
```

Comment:
I don't know if we want it here in the generic delete func...
Having it in the test cleanup should be good enough to prevent it from leaking again.

Reply:
If you really want to add it to the general cleanup, I'd add a RemoveTestFinalizers func and call it before the deletes.

Reply:
Agreed it shouldn't be necessary here. My thinking was that having it in Clean as well would help with recovering previously wedged clusters, but clusters shouldn't get wedged any more with the test-specific cleanup.

I'm not too attached to the generic cleanup, so I'll remove

Reply:
Done